src/client/dfs/dfs.c

```diff
@@ -1644,6 +1643,85 @@ dfs_readdir(dfs_t *dfs, dfs_obj_t *obj, daos_anchor_t *anchor, uint32_t *nr,
 	return rc;
 }
 
+int
+dfs_readdir_size(dfs_t *dfs, dfs_obj_t *obj, daos_anchor_t *anchor,
+		 uint32_t *nr, size_t size, dfs_readdir_cb_t op, void *udata)
+{
+	daos_key_desc_t	*kds;
+	d_sg_list_t	sgl;
+	d_iov_t		iov;
+	uint32_t	num, keys_nr;
+	char		*enum_buf;
+	int		rc;
+
+	if (dfs == NULL || !dfs->mounted)
+		return -DER_INVAL;
+	if (obj == NULL || !S_ISDIR(obj->mode))
+		return -DER_NOTDIR;
```

Comment:
Conflict with errno patch should be addressed.

Reply:
will do when it lands